internal/ingest/linksystem.go

```diff
@@ -225,24 +225,6 @@ func (ing *Ingester) ingestAd(publisher peer.ID, adCid cid.Cid) error {
 		skip = true
 	}
 
-	if !skip {
-		skipKey := string(contextID) + providerID.String()
-		skip = ing.hasSkip(skipKey)
-		if skip {
-			// An ad that is later in the chain has deleted all content for
-			// this ad, so skip this ad.
-			err = nil
-			log.Infow("Skipped advertisement that is removed later")
-		} else if isRm {
-			// This ad deletes all content for a contextID.  So, skip any
-			// previous (earlier in chain) ads that arrive later, that have
-			// the same contextID and provider.
-			ing.addSkip(skipKey)
```

Comment:
A skip key is only added if the ad is a removal.  Since we only do the removal by context ID, which removes all content for a context ID, it should be safe to skip any earlier ads that have the same context ID, regardless of what they do.

Reply:
Can handle in separate PR.